includes/polling/os/ray.inc.php

```diff
@@ -1,7 +1,40 @@
 <?php
-$ray_tmp = snmp_get_multi_oid($device, 'productName swVer serialNumber unitType', '-OQs', 'RAY-MIB');
-$hardware      = $ray_tmp['productName'];
-$version       = $ray_tmp['swVer'];
-$serial        = $ray_tmp['serialNumber'];
-$features      = $ray_tmp['unitType'];
+//Ray3
+$ray_tmp = snmp_get_multi_oid($device, 'productName.0 swVer.0 serialNumber.0 unitType.0', '-OQs', 'RAY-MIB');
+$hardware      = $ray_tmp['productName.0'];
+$version       = $ray_tmp['swVer.0'];
+$serial        = $ray_tmp['serialNumber.0'];
+$features      = $ray_tmp['unitType.0'];
+
+//Ray1 and Ray2 has no index
+if (empty($hardware)) {
+    $ray_tmp = snmp_get_multi_oid($device, 'productName swVer serialNumber unitType', '-OQs', 'RAY-MIB');
+    $hardware      = $ray_tmp['productName'];
+    $version       = $ray_tmp['swVer'];
+    $serial        = $ray_tmp['serialNumber'];
+    $features      = $ray_tmp['unitType'];
+
+    $snmpdata = snmp_get_multi_oid($device, ['sysName', 'sysObjectID', 'sysDescr'], '-OUQn', 'SNMPv2-MIB');
```

Comment:
This data is already fetched and in the $device array.

Reply:
I don't understand this. If I don't use this command, where do I get the data?

Reply:
@erotel This data is already polled and stored in $device. Just look at $device['sysName'], $device['sysObjectID'], $device['sysDescr'].

Reply:
Not the data is not there because Ray 1 and Ray 2 have bad snmp (do not support index)

Ray 2 
`snmpwalk -c public -v1 10.133.41.27 .1.3.6.1.2.1.1.5.0
iso.3.6.1.2.1.1.5 = STRING: "Zes-Dol30-24GHz"`

Ray 3
`snmpwalk -c public -v1 10.133.51.251 .1.3.6.1.2.1.1.5.0
iso.3.6.1.2.1.1.5.0 = STRING: "husak-achp_RAy3-24L"`

Reply:
Hmmmmmmmmmm, not sure the best way to fix, but I don't think this is it.  I hate bad SNMP implementations.

Reply:
I pointed out the poor implementation of snmp directly to the developer, unfortunately there was no fix. Couldn't you do that in the discovery section of the os folder? If this was a major problem, it would not be implemented.

Reply:
So I tested it. I propose to preserve os ray for Ray1, Ray2 and create os ray3 for Ray3. I would separate them with a file in `/includes/discovery/os` ,as ubnt has, then Ray3 could use a yaml file.